cdk/lib/prism-ec2-app.ts

```diff
@@ -0,0 +1,80 @@
+import type { CfnAutoScalingGroup } from "@aws-cdk/aws-autoscaling";
+import { BlockDeviceVolume, EbsDeviceVolumeType } from "@aws-cdk/aws-autoscaling";
+import { Peer } from "@aws-cdk/aws-ec2";
+import type { App } from "@aws-cdk/core";
+import { Duration } from "@aws-cdk/core";
+import { AccessScope, GuPlayApp } from "@guardian/cdk";
+import type { AppIdentity } from "@guardian/cdk/lib/constructs/core/identity";
+import type { GuStackProps } from "@guardian/cdk/lib/constructs/core/stack";
+import { GuStack } from "@guardian/cdk/lib/constructs/core/stack";
+import {
+  GuAllowPolicy,
+  GuAssumeRolePolicy,
+  GuDynamoDBReadPolicy,
+  GuGetS3ObjectsPolicy,
+} from "@guardian/cdk/lib/constructs/iam";
+
+export class PrismEc2App extends GuStack {
+  private static app: AppIdentity = {
+    app: "prism",
+  };
+
+  constructor(scope: App, id: string, props: GuStackProps) {
+    super(scope, id, props);
+
+    const pattern = new GuPlayApp(this, {
+      ...PrismEc2App.app,
+      userData: {
+        distributable: {
+          fileName: "prism.deb",
+          executionStatement: `dpkg -i /${PrismEc2App.app.app}/prism.deb`,
+        },
+      },
+      certificateProps: {
+        CODE: { domainName: "prism.code.dev-gutools.co.uk" },
+        PROD: { domainName: "prism.gutools.co.uk" },
+      },
+      monitoringConfiguration: { noMonitoring: true },
+      access: { scope: AccessScope.INTERNAL, cidrRanges: [Peer.ipv4("10.0.0.0/8")] },
+      roleConfiguration: {
+        additionalPolicies: [
+          new GuAllowPolicy(this, "DescribeEC2BonusPolicy", {
+            resources: ["*"],
+            actions: ["EC2:Describe*"],
+          }),
+          new GuDynamoDBReadPolicy(this, "ConfigPolicy", { tableName: "config-deploy" }),
+          new GuGetS3ObjectsPolicy(this, "DataPolicy", {
+            bucketName: "prism-data",
+          }),
+          new GuAssumeRolePolicy(this, "CrawlerPolicy", {
+            resources: ["arn:aws:iam::*:role/*Prism*", "arn:aws:iam::*:role/*prism*"],
+          }),
+        ],
+      },
+      scaling: {
+        CODE: { minimumInstances: 1 },
+        PROD: { minimumInstances: 2 },
+      },
+      blockDevices: [
+        {
+          deviceName: "/dev/sda1",
+          volume: BlockDeviceVolume.ebs(8, {
+            volumeType: EbsDeviceVolumeType.GP2,
+          }),
+        },
+      ],
+    });
+
+    // The pattern does not currently offer support for customising healthchecks via props
+    pattern.targetGroup.configureHealthCheck({
```

Comment:
This ensures that we keep:

https://github.com/guardian/prism/blob/b807073d2a2fad270b5576551772e46b81fd9fa5/cdk/lib/prism.ts#L115-L120

Which differs from [our default](https://github.com/guardian/cdk/blob/main/src/constructs/loadbalancing/alb/application-target-group.ts#L37-L44) significantly.

Reply:
Would be good to update the pattern to natively support custom healthchecks. One for later, I guess.

Reply:
Yes, possibly. It's hard to tell how much of the variation here is due to careful tuning vs historical guesswork though!